src/nl/rubensten/texifyidea/settings/TexifySettings.kt

```diff
@@ -24,9 +25,50 @@ class TexifySettings : PersistentStateComponent<TexifySettings> {
     var automaticUpDownBracket = true
     var automaticItemInItemize = true
 
+    /**
+     * internal list which stores the commands data
+     */
+    private val _labelCommands: HashMap<String, LabelingCommandInformation> =
+            hashMapOf("\\label" to LabelingCommandInformation("\\label", 1, true))
+
+    /**
+     * Verify that any key in the list has a leading backslash
+     */
+    val labelCommands: Map<String, LabelingCommandInformation>
+    get() {
+        return _labelCommands.mapKeys { it.key.addLeadingSlash() }
+    }
+
+    /**
+     * Do not use this field anywhere! It is only needed to persist the data of the commands to the settings.
+     * It maps any command to its string representation and on first load back to the object
+     */
+    var labelCommandsAsString: Map<String, String>
```

Comment:
It is not used? Why is it here then? This is probably not the correct way of doing this. I'd suggest handling the input in dedicated accessor functions (like `addCommand`?) to abstract the process.

Reply:
It is used by IntelliJ's function to store settings. IntelliJ could only store simple lists or maps with simple types. As I need to store a complex type this field maps the complex type to a string representation which could be stored by IntelliJ.
I hope you get what I wanted to do. If you have a better idea for this case, please let me hear it.

Reply:
So, if I understand this correctly (and yes I've read the docs :P), this member's `set` gets called whenever the serialiser sets this property, thus automatically altering the state of the backing field?

This does feel like property abuse to be honest. I'd suggest just having a plain `var labelCommandsAsString` and modify it in `loadState` and `getState` to produce the serialised versions, so you separate state and behaviour a bit.

Reply:
Well, the `get` gets called every time the serialiser write the information to the `settings.xml` file and the `set` is called every time the deserialiser reads from the file.
I did it this way, because I don't want to store the same information twice. If there is only one field, it is in the same state every time, if there is mor than one field, you have to update both and when you don't to that, there is an inconsistency...
And there was already the backing field so it was the easiest way to reuse it and only remap the properties.
What do you mean exactly with "seperate state and behaviour"? The state is saved in the backing field and the behaviour is done by the diffrent properties calling to it.

Reply:
@Stainzl @RubenSchellekens There is actually a way to separate state and behavior here. See the first example in the docs: http://www.jetbrains.org/intellij/sdk/docs/basics/persisting_state_of_components.html?search=state#implementing-the-persistentstatecomponent-interface

Create a State class for serialization with the Map<String, String> object.
In the example, use a different type for `myState` to include the Map<String, LabelingCommandInformation> type, and handle the conversion in getState/loadState, where it semantically belongs. This avoids property abuse.

Note that since we only used 'simple' settings properties before it was not necessary to use a separate `State` class before. It will require some refactoring to migrate the other settings as well. Let me know if you need some help with that.

Reply:
Okay, I hope I understood it right. I tried it in commit 0e35e31a1ad4c4f5d00571372c8edc8f15ec1626. Could you review it and give me feedback?